spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/procedures/RewriteDataFilesProcedure.java

```diff
@@ -134,6 +138,17 @@ private RewriteDataFiles checkAndApplyFilter(
     return action;
   }
 
+  private RewriteDataFiles checkAndApplyBranch(
+      Table table, Identifier ident, RewriteDataFiles action) {
+    String branchIdent = Spark3Util.extractBranch(ident);
+    SparkWriteConf writeConf = new SparkWriteConf(spark(), table, branchIdent, Maps.newHashMap());
```

Comment:
SparkWriteConf should not be initialized independently within this method, it feels like it should be a part of `RewriteDataFilesProcedure` class, or even the `BaseProcedure` class.

Reply:
probably good to at least initialize it at RewriteDataFilesProcedure level, and we can see if it can be shared across procedures in a later PR.